### PR TITLE
Remove some signed/unsigned integer comparison warnings

### PR DESCRIPTION
**Pull request by sdenel**

### CODE REVIEW

**Comment by zheng-xq** [comment]
Adding @keveman for review. 

However, I think TensorFlow in general prefers signed int over unsigned int. But I'll leave it up to @keveman.

**Comment by sdenel** [comment]
Changes mainly look like the ones from this pull request, accepted at the time: https://github.com/tensorflow/tensorflow/pull/1099

**Comment by rmlarsen** [comment]
@tensorflow-jenkins test this please.

File: tensorflow/core/common_runtime/executor.cc [hunk 1]
```diff
@@ -724,7 +724,7 @@ class ExecutorState {
 
    private:
     gtl::InlinedVector<TaggedNode, 16> ready_;
-    int front_index_;
+    size_t front_index_;
```
Comment (rmlarsen) [inline tensorflow/core/common_runtime/executor.cc]: I would leave this one alone.

**Comment by rmlarsen** [comment]
@sdenel it looks like some of these were already fixed. Can you please rebase your PR? In general, this PR looks fine to me apart from the change to ExecutorState.

**Comment by sdenel** [comment]
@rmlarsen I just rebased my PR.

**Comment by rmlarsen** [comment]
@tensorflow-jenkins test this please

**Comment by rmlarsen** [comment]
@sdenel It looks like you have to rebase again to pick up the changes I pushed earlier to day, sorry about the duplicate work.

**Comment by sdenel** [comment]
@rmlarsen done!

**Comment by rmlarsen** [comment]
@sdenel I'm still seeing a bunch of changes to the shape inference code in array_ops.cc. did you fail to resolve a merge conflict?

**Comment by sdenel** [comment]
@rmlarsen It looks like I failed my first rebase :) Sorry for the mess, it should ok now.

**Comment by rmlarsen** [comment]
@tensorflow-jenkins test this please

@sdenel Thanks!

**Comment by rmlarsen** [comment]
@tensorflow-jenkins test this please.

**Comment by rmlarsen** [comment]
@sdenel thanks for the cleanup!

**Pull request merged**